src/libraries/Microsoft.Extensions.Caching.Memory/src/MemoryCache.cs

```diff
@@ -78,7 +74,7 @@ public MemoryCache(IOptions<MemoryCacheOptions> optionsAccessor, ILoggerFactory
         public int Count => _entries.Count;
 
         // internal for testing
-        internal long Size { get => Interlocked.Read(ref _cacheSize); }
+        internal long Size => Volatile.Read(ref _cacheSize);
```

Comment:
Is this a "safe" change? Are there considerations for other architectures beyond x64?

Reply:
Yes, interlocking doesn't add any safety here - it's just a snapshot of the value for test assertions.

Reply:
The updated logic in `UpdateCacheSizeExceedsCapacity` makes the uncontested case faster because it does an optimistic volatile read first and doesn't uses the result of `Interlocked.CompareExchange` instead of doing a second `Interlocked.Read` operation.